libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -226,6 +226,22 @@ bool SyntaxChecker::visit(FunctionDefinition const& _function)
 				"\"."
 			);
 	}
+
+	if (_function.isOldStyleConstructor())
+	{
+		if (v050)
+			m_errorReporter.syntaxError(
+				_function.location(),
+				"Functions are not allowed to have the same name as the contract. "
```

Comment:
Slightly unrelated, but I guess we could lift this restriction in the future (couple of releases into 0.5.0). It shouldn't matter what the function name is at that point.

Reply:
I'm not sure if we should lift this restriction, it will result in really different contracts that might also be extremely dangerous (think parity hack). If you use the recommended style, this should also not be a problem, since functions are lowercase, contracts are uppercase.